prdoc/pr_5743.prdoc

```diff
@@ -0,0 +1,22 @@
+title: "[pallet-revive] write sandbox output according to the provided output buffer length"
+
+doc:
+  - audience: Runtime Dev
+    description: |
+     Instead of error out if the provided output buffer is smaller than what we want to write,
+     we can just write what fits into the output buffer instead.
+     We already write back the actual bytes written to the in-out pointer,
+     so contracts can check it anyways.
+
+     This in turn introduces the benefit of allowing contracts to implicitly request only a portion
+     of the returned data from calls and incantations.
+     Which is especially beneficial for YUL as the call family opcodes have a return data size
+     argument and this change removes the need to work around it in contract code.
```

Comment:
Not too sure to get the point here

Reply:
YUL sometimes want to ignore (parts) of the output but still learn about the size. This was not possible with our code. You could even ignore everything (supply sentinel) but then also don't learn about the length or you have to read everything.